packages/query/src/index.ts

```diff
@@ -1488,7 +1488,15 @@ ${override.query.shouldExportQueryKey ? 'export ' : ''}const ${queryOption.query
         ? `'infinite'`
         : '',
       queryKeyIdentifier,
-      queryParams ? '...(params ? [params]: [])' : '',
+      props
```

Comment:
Shouldn't this complex logic be moved above as a `const` and explained with a comment?

Reply:
Done :+1: